rls/src/build/cargo.rs

```diff
@@ -226,15 +203,15 @@ fn run_cargo_ws(
             opts.all_targets,
         ),
         build_config: BuildConfig::new(
-            &config,
+            &ws.config(),
             opts.jobs,
             &opts.target,
             CompileMode::Check { test: cfg_test },
         )?,
         features: opts.features,
         all_features: opts.all_features,
         no_default_features: opts.no_default_features,
-        ..CompileOptions::new(&config, CompileMode::Check { test: cfg_test })?
+        ..CompileOptions::new(&ws.config(), CompileMode::Check { test: cfg_test })?
```

Comment:
This (and `ws.config()` above) is wrong - we need to pass our custom `CargoConfig` that's created before at https://github.com/rust-lang/rls/pull/1483/files#diff-733fe8366c72147252e30946db56b0e7L130